### PR TITLE
Enhance LogUtils

### PR DESCRIPTION
**Pull request by amcclain**
+ Optionally accept messages as an array of any objects. Continue to concat string messages together with standardized | separator, log any other objects in their natural form.
+ Add logError() and logWarn()
+ Convert (some) existing console.log/warn/error calls to use the new utils - we should sweep through and update the rest of them before merge.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [ ] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by amcclain** [comment]
For review/discussion - if we want to take this change, can add CHANGELOG entry + we should sweep through and upgrade other raw calls to console loggers where appropriate.

File: utils/js/LogUtils.ts [hunk 1]
```diff
@@ -103,7 +121,51 @@ function parseSource(source) {
     return '';
 }
 
-function writeLog(msg, source, level) {
-    if (source) msg = `[${source}] ${msg}`;
-    level === 'info' ? console.log(msg) : console.debug(msg);
+function writeLog(msgs: Some<any>, source, level: LogLevel) {
```
Comment (lbwexler) [inline utils/js/LogUtils.ts:124]: this is alway an array, fwiw

File: utils/js/LogUtils.ts [hunk 2]
```diff
@@ -20,60 +21,77 @@ import {castArray, isString} from 'lodash';
  * @param fn - function to execute
  * @param source - class, function or string to label the source of the message
  */
-export function withInfo<T>(msgs: string[] | string, fn: () => T, source?: any): T {
+export function withInfo<T>(msgs: Some<any>, fn: () => T, source?: any): T {
```
Comment (lbwexler) [inline utils/js/LogUtils.ts:24]: Think we should type these as Some<unknown>

File: utils/js/LogUtils.ts [hunk 3]
```diff
@@ -103,7 +121,51 @@ function parseSource(source) {
...
-    if (source) msg = `[${source}] ${msg}`;
-    level === 'info' ? console.log(msg) : console.debug(msg);
+function writeLog(msgs: Some<any>, source, level: LogLevel) {
+    msgs = castArray(msgs);
+    if (source) msgs = [`[${source}]`, ...msgs];
+
+    const logArgs = [];
+    let strMsg;
+
+    msgs.forEach((msg, idx) => {
```
Comment (lbwexler) [inline utils/js/LogUtils.ts:131]: I think this all would be a bit easier to build logArgs with reduce.  With every step you get the current state of the array and then the next element. For each step,  either push a "|" and your arg, or just append your arg to the string that is the last element

File: utils/js/LogUtils.ts (hunk 2 above)
Comment (amcclain) [reply utils/js/LogUtils.ts]: Changed signature on public APIs. Ran into type errors if I changed impl methods that I did not understand.

File: utils/js/LogUtils.ts (hunk 1 above)
Comment (amcclain) [reply utils/js/LogUtils.ts]: Updated to `msgs: Array<any>` and removed castArray

File: utils/js/LogUtils.ts (hunk 3 above)
Comment (amcclain) [reply utils/js/LogUtils.ts:130]: Sorry couldn't follow, tried but did not find an easier path with reduce. I tend not to reach for reduce as I almost never find it intuitive or easy to read. More than happy to have a different impl here, just ran out of time to figure it out myself.

**Comment by amcclain** [comment]
Replaced by #3534

**Pull request closed without merging**